### PR TITLE
[issue-2916] Add DeliveryPolicy and FilterPolicy to SNS resource

### LINKED ISSUE #2916: Enhancement: Expose "Subscription Filter Policy" for AWS SNS Topic Subscriptions
_This issue was originally opened by @rbwestmoreland as hashicorp/terraform#17072. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

### Enhancement

* Provider: `aws`
* Resource: `aws_sns_topic_subscription`

Please expose the "Subscription Filter Policy" feature for the `aws_sns_topic_subscription` resource. 

> A subscription filter policy is a simple JSON object that enables your subscriber to receive only the subset of messages it’s actually interested in, as opposed to receiving every single message published to the topic.
> 
> When a message is published to the topic, Amazon SNS will attempt to match the incoming message attributes to the subscription attribute that defines a filter policy. If they match, Amazon SNS will then deliver the message to the corresponding subscriber.

### References
* New Feature Announcement: <https://aws.amazon.com/blogs/compute/simplify-pubsub-messaging-with-amazon-sns-message-filtering/>  
* AWS Console Example (See "Step 4"): <https://aws.amazon.com/getting-started/tutorials/filter-messages-published-to-topics/>
* AWS Docs: <https://docs.aws.amazon.com/sns/latest/dg/message-filtering.html>

### PR DESCRIPTION
**Pull request by ddcprg**
Fixes #2916 

I'll try to add some acceptance tests for each of the attributes.

### CODE REVIEW

File: aws/resource_aws_sns_topic_subscription_test.go [hunk 1]
```diff
@@ -31,6 +31,27 @@ func TestAccAWSSNSTopicSubscription_basic(t *testing.T) {
...
+					testAccCheckAWSSNSTopicExists("aws_sns_topic.test_topic"),
+					testAccCheckAWSSNSTopicSubscriptionExists("aws_sns_topic_subscription.test_subscription"),
+					resource.TestCheckResourceAttr("aws_sns_topic_subscription.test_subscription", "filter_policy",
+						"{\"store\": [\"example_corp\"], \"event\": [\"order_placed\"]}"),
+				),
+			},
+		},
+	})
+}
+
```
Comment (ddcprg) [inline aws/resource_aws_sns_topic_subscription_test.go]: I'm afraid I cannot test `delivery_policy` in the subscription. It should be easy, just need to add the attribute in one of the HTTP endpoint test, something like

`delivery_policy        = "{\"healthyRetryPolicy\": {\"numRetries\": 51}}"`

and then assert in the acceptance test:

```
resource.TestCheckResourceAttr("aws_sns_topic_subscription.test_subscription", "delivery_policy", "{\"healthyRetryPolicy\": {\"numRetries\": 51}}"),
```

**Comment by bflad** [comment]
@ddcprg as a heads up, `filter_policy` was previously submitted as a PR in #2806 and will be merged ahead of this one as we generally prefer reviewing and accepting earlier contributions. Likely this PR will need to be rebased off of that one, but you might want to wait until its done. The `delivery_policy` functionality here will definitely be appreciated!

**Comment by ddcprg** [comment]
@bflad that's fine, I didn't see the other PR. I'll take a look at the other PR and see if I can push the `dlivery_policy` bits against that code in a separate PR. Cheers

**Comment by ddcprg** [comment]
I'm closing this and opening #3289

**Pull request closed without merging**